packages/shorebird_cli/lib/src/executables/aot_tools.dart

```diff
@@ -98,21 +109,68 @@ class AotTools {
     // distributed as an executable.
     final extension = p.extension(artifactPath);
     if (extension != '.dill' && extension != '.dart') {
-      return process.run(
+      return runFn(
         artifactPath,
         command,
         workingDirectory: workingDirectory,
       );
     }
 
     // local engine versions use .dart and we distribute aot-tools as a .dill
-    return process.run(
+    return runFn(
       shorebirdEnv.dartBinaryFile.path,
       ['run', artifactPath, ...command],
       workingDirectory: workingDirectory,
     );
   }
 
+  /// Similar to [_exec], but logs the sub process stdout and stderr
+  /// as they are emmited.
+  Future<ShorebirdProcessResult> _execWithLiveLogs(
+    List<String> command, {
+    String? workingDirectory,
+  }) {
+    return _exec(
+      command,
+      runCommand: (
+        String exe,
+        List<String> args, {
+        String? workingDirectory,
+      }) async {
+        final spawnedProcess = await process.start(
+          exe,
+          args,
+          workingDirectory: workingDirectory,
+        );
+
+        final stdout = StringBuffer();
+        final stderr = StringBuffer();
+
+        final stdoutSub = spawnedProcess.stdout.map(utf8.decode).listen((data) {
+          logger.detail(data);
+          stdout.write(data);
```

Comment:
This reads like `data` will be printed twice when running in verbose mode. Is that not the case?

Reply:
Nope, because nobody uses the stdout of the result. That was not the case for the stderr, I adjusted that.

Good catch!

I will improve this more when we move this up to the shorebird process class.